src/helm/benchmark/scenarios/melt_lm_scenarios.py

```diff
@@ -0,0 +1,250 @@
+        return instances
+
+
+class MELTLMMaskFillingMLQAScenario(MELTLMMaskFillingScenario):
+    """
+    Scenario for the MLQA dataset.
+    This dataset is a multilingual question answering dataset.
+    It contains questions in multiple languages and their corresponding
+    answers in the same language. In this scenario, we are using the
+    context of questions in the Vietnamese subset of the MLQA dataset.
+    """
+
+    name = "melt_lm_mask_filling_mlqa"
+    description = "MLQA dataset for masked language modeling."
+    tags = ["language_modeling", "mask_filling"]
+
+    def __init__(self):
+        super().__init__(
+            dataset_name="facebook/mlqa",
+            revision="397ed406c1a7902140303e7faf60fff35b58d285",
+            subset="mlqa.vi.vi",
+            text_key="context",
+            splits={
+                VALID_SPLIT: "validation",
+                TEST_SPLIT: "test",
+            },
+        )
+
+
+class MELTLMSpellingCorrectionScenario(Scenario):
+    """
+    Scenario for the MELT spelling correction dataset.
+    """
+
+    name = "melt_lm_spelling_correction"
+    description = "Spelling Correction scenario."
+    tags = ["language_modeling", "spelling_correction"]
+
+    def __init__(
+        self,
+        dataset_name: str,
+        revision: str,
+        source_key: str = "text",
+        target_key: str = "corrected_text",
+        subset: Optional[str] = "",
+        splits: Optional[Dict[str, str]] = None,
+    ):
+        """Initializes the question answering scenario.
+
+        Args:
+            dataset_name: The name of the dataset.
+            revision: The revision of the dataset to use.
+            source_key: The key to use for the source text in the dataset. Defaults to "text".
+            target_key: The key to use for the target text in the dataset. Defaults to "corrected_text".
+            subset: The subset of the dataset to use. Defaults to "".
+            splits: The splits to use for the dataset. Defaults to None.
+        """
+        super().__init__()
+        self.dataset_name = dataset_name
+        self.subset = subset
+        self.source_key = source_key
+        self.target_key = target_key
+        self.revision = revision
+        self.splits = splits
+
+    def get_instances_for_splits(self, splits: Dict[str, str]) -> List[Instance]:
+        """
+        Helper for generating instances for a split.
+        Args:
+            splits (dict): Which splits to partition the data into.
+        Returns:
+            List[Instance]: Instances from the file for the specified split.
+        """
+        instances: List[Instance] = []
+        dataset: Any = load_dataset(
+            self.dataset_name,
+            self.subset,
+            revision=self.revision,
+            trust_remote_code=True,
+        )
+        if len(splits) == 1:
+            all_keys = list(splits.keys())
+            dataset = dataset[all_keys[0]].train_test_split(test_size=0.33)
```

Comment:
Need to set the `seed` parameter to get deterministic results ([doc](https://huggingface.co/docs/datasets/v3.6.0/en/package_reference/main_classes#datasets.Dataset.train_test_split))